src/include/miopen/conv/heur/heur.hpp

```diff
@@ -0,0 +1,173 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2020 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#ifndef GUARD_MIOPEN_HEUR_HPP
+#define GUARD_MIOPEN_HEUR_HPP
+#include <miopen/handle.hpp>
+#include <miopen/convolution.hpp>
+#include <miopen/any_solver.hpp>
+#include <miopen/conv/heur/metadata.hpp>
+#include <miopen/anyramdb.hpp>
+#include <boost/filesystem.hpp>
+#include <nlohmann/json.hpp>
+#include <fstream>
+#include <string>
+#include <math.h>
+#include <algorithm>
+
+namespace miopen {
+
+struct ConvHeur
+{
+    static const nlohmann::json& GetMetadata(const std::string& arch)
+    {
+        static const nlohmann::json metadata = nlohmann::json::parse(
+            std::ifstream(boost::filesystem::path(GetSystemDbPath() + arch + "_metadata.model")
+                              .generic_string()));
+        return metadata;
+    }
+
+    static const std::vector<std::string>& GetSupportedArchs()
+    {
+        static const std::vector<std::string> supported_archs = {"gfx908"};
+        return supported_archs;
+    }
+
+    static bool IsApplicable(const std::string& arch, const conv::ProblemDescription& problem)
+    {
+        MIOPEN_LOG_I2("");
```

Comment:
[Recommendation][Whole PR] Please reconsider the added logging. Maybe it is possible to remove some of logging (or lower its level) when it is not really important for debugging.